src/vivarium_conic_lsff/components/fortification/folic_acid.py

```diff
@@ -155,15 +162,116 @@ def adjust_birth_prevalence(self, index, birth_prevalence):
     def load_relative_risk_data(builder: 'Builder') -> float:
         location = builder.configuration.input_data.location
         draw = builder.configuration.input_data.input_draw_number
-        return sample_folic_acid_relative_risk(location, draw)
+        return params.sample_folic_acid_relative_risk(location, draw)
 
     @staticmethod
     def load_population_attributable_fraction_data(builder: 'Builder'):
         location = builder.configuration.input_data.location
         draw = builder.configuration.input_data.input_draw_number
-        rr = sample_folic_acid_relative_risk(location, draw)
-        coverage = sample_folic_acid_coverage(location, draw, 'baseline')
+        rr = params.sample_folic_acid_relative_risk(location, draw)
+        coverage = params.sample_folic_acid_coverage(location, draw, 'baseline')
         exposure = 1 - coverage
         mean_rr = rr*exposure + 1*(1-exposure)
         paf = (mean_rr - 1)/mean_rr
         return paf
+
+
+class MaternalIronFortificationEffect:
+
+    @property
+    def name(self):
+        return 'maternal_iron_fortification_effect'
+
+    def setup(self, builder: 'Builder'):
+        self.treatment_effects = self.load_treatment_effects(builder)
+
+        builder.value.register_value_modifier(f'low_birth_weight_and_short_gestation.exposure',
+                                              self.adjust_birth_weights)
+
+        self._fortified = project_globals.IRON_FORTIFICATION_COVERAGE_COLUMN
+        self._amount_fortified_food = project_globals.IRON_FORTIFICATION_FOOD_CONSUMPTION
+        self.population_view = builder.population.get_view([self._fortified, self._amount_fortified_food])
+
+    def adjust_birth_weights(self, index, birth_weights):
+        fortified_status = self.population_view.get(index)[self._fortified]
+        idx_fortified = fortified_status.loc[fortified_status=='covered'].index
+        if len(idx_fortified):
+            amount_fortified_food = self.population_view.get(index)[self._amount_fortified_food]
+            bw_shift = self.compute_shift(amount_fortified_food.loc[idx_fortified])
+            birth_weights.loc[idx_fortified, 'birth_weight'] += bw_shift
+        return birth_weights
+
+    def compute_shift(self, amount_fortified_food: pd.Series) -> pd.Series:
+        location, draw, amount_compound_iron, iron_effect = self.treatment_effects
+
+        elemental_iron = amount_compound_iron * params.IF_ELEMENTAL_IRON_RATIO
+        iron_per_day = params.gram_to_kg(amount_fortified_food) * elemental_iron
+
+        seed = get_hash(f'iron_fortification_coverage_draw_{draw}_location_{location}')
```

Comment:
This pattern is only for initialization.  Build a frozen distribution, sample with ppf with a draw from the randomness stream